File: src/volumedriver/pythontest/toolcut_test.py
# Copyright (C) 2016 iNuron NV
#
# This file is part of Open vStorage Open Source Edition (OSE),
# as available from
#
#      http://www.openvstorage.org and
#      http://www.openvstorage.com.
#
# This file is free software; you can redistribute it and/or modify it
# under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
# as published by the Free Software Foundation, in version 3 as it comes in
# the LICENSE.txt file of the Open vStorage OSE distribution.
# Open vStorage is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY of any kind.

import os
import volumedriver.toolcut.ToolCut as T
import volumedriver.toolcut.backend as Backend
from volumedriver.base import env
import nose


THISDIR = os.path.dirname(os.path.abspath(__file__))
VOLNAME = "5c17a9f4f16348b8bbc"

logger = env.getSublogger(__name__)


@nose.tools.nottest
def testSP(target):
    sp = T.SnapshotPersistor(target.params(), target.namespace())
    count = 0
    for s in sp.getSnapshots():
        logger.info(s.name())
        logger.info(s.stored())
        logger.info(s.date())
        nose.tools.eq_(s.stored(), sp.snapshotStored(s.name()))
        count += s.stored()
    count += sp.currentStored()
    nose.tools.eq_(sp.stored(), count)


@nose.tools.nottest
def testSnapshotPersistorToolCut():
    testSP(Backend.LocalBackend(os.path.join(THISDIR, "data"))(VOLNAME))
    testSP(Backend.LocalBackend(os.path.join(THISDIR, "data"))
           (VOLNAME + "-version3"))
    testSP(Backend.LocalBackend(os.path.join(THISDIR, "data"))
           (VOLNAME + "-version4"))

File: src/volumedriver/test/ToolCutTest.cpp
// Copyright (C) 2016 iNuron NV
//
// This file is part of Open vStorage Open Source Edition (OSE),
// as available from
//
//      http://www.openvstorage.org and
//      http://www.openvstorage.com.
//
// This file is free software; you can redistribute it and/or modify it
// under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
// as published by the Free Software Foundation, in version 3 as it comes in
// the LICENSE.txt file of the Open vStorage OSE distribution.
// Open vStorage is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY of any kind.

#include "VolManagerTestSetup.h"

namespace volumedriver
{
class ToolCutTest
    : public VolManagerTestSetup
{
public:
    ToolCutTest()
        : VolManagerTestSetup("ToolCutTest")
    {}
};

TEST_P(ToolCutTest, DISABLED_test1)
{
    VolumeId vid("volume");
    backend::Namespace ns;
    SharedVolumePtr v = newVolume(vid,
                          ns);

    const std::string pattern("immanuel");
    sleep(1000);
    destroyVolume(v,
                  DeleteLocalData::T,
                  RemoveVolumeCompletely::T);
}

TEST_P(ToolCutTest, DISABLED_test2)
{
    VolumeId vid("volume");
    backend::Namespace ns;
    SharedVolumePtr v = newVolume(vid,
                          ns);

    const SnapshotName snapshot("snapshot");
    createSnapshot(*v,
                   snapshot);

    waitForThisBackendWrite(*v);

    SharedVolumePtr c = createClone("clone",
                            backend::Namespace(),
                            ns,
                            snapshot);

    const std::string pattern("immanuel");
    sleep(1000);
    destroyVolume(c,
                  DeleteLocalData::T,
                  RemoveVolumeCompletely::T);
    destroyVolume(v,
                  DeleteLocalData::T,
                  RemoveVolumeCompletely::T);


}

TEST_P(ToolCutTest, DISABLED_toolcut_create_volume)
{
    backend::Namespace ns1;

    SharedVolumePtr v = newVolume(VolumeId("volume1"),
                          ns1);

    writeToVolume(*v, Lba(0), 4096,"immanuel");

    const SnapshotName snap1("snap1");

    v->createSnapshot(snap1);

    waitForThisBackendWrite(*v);
    sleep(100000);

}

INSTANTIATE_TEST(ToolCutTest);

}
// Local Variables: **
// mode: c++ **
// End: **
